src/dotty/tools/dotc/core/Names.scala

```diff
@@ -221,13 +221,15 @@ object Names {
   private var size = 1
 
   /** The hash of a name made of from characters cs[offset..offset+len-1].  */
-  private def hashValue(cs: Array[Char], offset: Int, len: Int): Int =
-    if (len > 0)
-      (len * (41 * 41 * 41) +
-        cs(offset) * (41 * 41) +
-        cs(offset + len - 1) * 41 +
-        cs(offset + (len >> 1)))
-    else 0
+  private def hashValue(cs: Array[Char], offset: Int, len: Int): Int = {
+    var i = offset
+    var hash = 0
+    while (i < len) {
```

Comment:
Don't think so: https://github.com/scala/scala/pull/5474/files#diff-dd6d9f33ec7fb0a0b04f787bdb442bbdR47. We're already offsetting in `i`.